From 14b3f63ef8611b9870ad76dbb657b533e40b0dbe Mon Sep 17 00:00:00 2001
From: josselinonduty <contact@josselinonduty.fr>
Date: Mon, 1 Dec 2025 13:28:30 +0100
Subject: [PATCH] feat: add toggle to hide application offline banner
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Original patch: Aurélien Hamy <aunetx@yandex.com>
This solves #123 by adding the `--hide-offline-banner` command-line argument
to hide the annoying "Application is offline" banner, that sometimes appears when
using a VPN or DNS level blocker.
---
 build/main.js | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/build/main.js b/build/main.js
index c0af7d5..cf4b722 100644
--- a/build/main.js
+++ b/build/main.js
@@ -3532,6 +3532,17 @@
           this.window.on("unmaximize", () => {
             this.window.webContents.send("window-maximize-changed", !1);
           }));
+      if (
+        process.argv.some((arg) => arg === "--hide-offline-banner") ||
+        "yes" === process.env.DZ_HIDE_OFFLINE_BANNER
+      )
+        this.appService.getWebContents().on("dom-ready", () => {
+          this.appService
+            .getWebContents()
+            .insertCSS(
+              `.alert-wrapper:has(> div[data-testid="alert-AppOffline"]) { display: none !important; }`
+            );
+        });
       const showTimeout = setTimeout(() => {
         !this.window ||
           this.window.isDestroyed() ||
-- 
2.53.0

